From 9967bbe9e3aa507c3fda39cc387ac73c834d8128 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Tue, 30 Jan 2024 16:41:58 +0100
Subject: [PATCH 408/463] arm64: dts: rk3566-pientab2: Use non-linear
 brightness->PWM duty cycle curve

This helps with dumb use of sysfs to control backlight brightness.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/rockchip/rk3566-pinetab2.dtsi | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pinetab2.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-pinetab2.dtsi
index db40281eafbe5..1ea2513eaafd8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-pinetab2.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-pinetab2.dtsi
@@ -44,9 +44,14 @@ button-vol-down {
 	backlight: backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm4 0 25000 0>;
-		brightness-levels = <20 220>;
-		num-interpolated-steps = <200>;
-		default-brightness-level = <100>;
+		brightness-levels =
+			<0 250 360 470 580 690 810 949 1110 1294 1502
+			1737 1998 2289 2610 2964 3351 3774 4233 4731
+			5268 5847 6467 7133 7845 8604 9412 10271 11182
+			12146 13164 14239 15374 16568 17822 19140 20521
+			21969 23483 25068 26722 28447 30247 32121 34071
+			36099 38210 40400 42669 45026 47468 50000>;
+		default-brightness-level = <17>;
 		power-supply = <&vcc_sys>;
 	};
 
-- 
2.46.0

